src/quo/components/wallet/account_origin/schema.cljs

```diff
@@ -0,0 +1,30 @@
+(ns quo.components.wallet.account-origin.schema)
+
+(def ^:private ?base
+  [:map
+   [:type {:optional true} [:enum :default-keypair :recovery-phrase :private-key]]
+   [:stored {:optional true} [:enum :on-device :on-keycard]]
+   [:theme :schema.common/theme]])
+
+(def ^:private ?default-keypair
+  [:map
+   [:user-name {:optional true} [:maybe :string]]
+   [:profile-picture {:optional true} [:maybe [:or :schema.common/image-source :string]]]
+   [:derivation-path {:optional true} [:maybe :string]]
+   [:on-press {:optional true} [:maybe fn?]]
+   [:customization-color {:optional true} [:maybe :schema.common/customization-color]]])
+
+(def ^:private ?recovery-phrase
+  [:map
+   [:derivation-path {:optional true} [:maybe :string]]
+   [:on-press {:optional true} [:maybe fn?]]])
+
+(def ?schema
+  [:=>
+   [:catn
```

Comment:
Not a suggestion for change @mmilad75, but rather a perception to incite possible guidelines.

Maybe we should prefer `cat` (not `catn`) for function schemas with just one argument, as is the case for most components. It adds an extra indentation and when these schemas fail it will be just as clear why. One might say it's nice for consistency because `catn` leads to more readable errors, but to me it seems superfluous to use it.

Maybe the best is to just keep the option so the developer use whatever they prefer, but we shouldn't feel forced to use `catn` all the time IMO.